medusa/storage/__init__.py

```diff
@@ -72,39 +66,36 @@ def __init__(self, *, config):
         self._k8s_mode = evaluate_boolean(config.k8s_mode) if config.k8s_mode else False
         self._prefix = pathlib.Path(config.prefix or '.')
         self.prefix_path = str(self._prefix) + '/' if len(str(self._prefix)) > 1 else ''
-        self.storage_driver = self._connect_storage()
+        self.storage_driver = self._load_storage()
         self.storage_provider = self._config.storage_provider
 
-    def _connect_storage(self):
+    def __enter__(self):
+        self.storage_driver.connect()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.storage_driver.disconnect()
+
+    def _load_storage(self):
         logging.debug('Loading storage_provider: {}'.format(self._config.storage_provider))
-        if self._config.storage_provider == Provider.GOOGLE_STORAGE:
+        if self._config.storage_provider.lower() == 'google_storage':
```

Comment:
**Suggestion:** we should create a refactoring ticket to create new constants for the storage providers.

Reply:
https://github.com/thelastpickle/cassandra-medusa/issues/646